### PR TITLE
Fix operator ports for metrics endpoint

### PR DESCRIPTION
**Pull request by periklis**
### Description
This PR is a cherry-pick of openshift/elasticsearch-operator#652 to ensure porting fix forward for https://bugzilla.redhat.com/show_bug.cgi?id=1924657.

This is not an automatic cherry-pick because the approach to adapt the CSV and required k8s resources to expose metrics port and service monitors is different to what we used in pre operator-sdk 1.x times. Since 5.0 we use operator-sdk 1.2 and in turn the new kustomize-based machinery to apply changes into the operators CSV.

/cc @huikang 
/assign @ewolinetz 

### Links
- Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1924657
- Depending PRs: #665

### CODE REVIEW

File: config/rbac/auth_proxy_service.yaml [hunk 1]
```diff
@@ -1,14 +1,16 @@
...
 metadata:
+  annotations:
+    service.beta.openshift.io/serving-cert-secret-name: elasticsearch-operator-metrics
   labels:
-    control-plane: controller-manager
-  name: controller-manager-metrics-service
-  namespace: system
+    name: elasticsearch-operator
+  name: elasticsearch-operator-metrics
+  namespace: openshift-operators-redhat
```
Comment (huikang) [inline config/rbac/auth_proxy_service.yaml:9]: Same here. We can customize the namespace from the kustomization.yaml

File: bundle/manifests/elasticsearch-operator-metrics-monitor_monitoring.coreos.com_v1_servicemonitor.yaml [hunk 2]
```diff
@@ -6,10 +6,15 @@ metadata:
...
+    interval: 30s
     path: /metrics
-    port: http
-    scheme: http
+    scheme: https
+    scrapeTimeout: 10s
+    targetPort: 8443
+    tlsConfig:
+      caFile: /etc/prometheus/configmaps/serving-certs-ca-bundle/service-ca.crt
+      serverName: elasticsearch-operator-metrics.openshift-operators-redhat.svc
```
[the remark this answers is not in the document]
Comment (periklis) [reply bundle/manifests/elasticsearch-operator-metrics-monitor_monitoring.coreos.com_v1_servicemonitor.yaml:17]: After some investigation I came to the unfortunate conclusion that we cannot use the kustomize `vars` feature for such a case. The `vars` substitution works only on a handful of fields - hardcoded in kustomize AFAICS - and ServiceMonitor is none of them. :(

File: config/rbac/auth_proxy_service.yaml (hunk 1 above)
Comment (periklis) [reply config/rbac/auth_proxy_service.yaml]: Good catch!

**Comment by ewolinetz** [comment]
nit, this doesn't seem like a cherrypick of the prior pr. can you give a tl;dr in the pr description as to what was wrong since there are 29 files here?

/approve

**Pull request merged**